app/lib/shared/configuration.dart

```diff
@@ -130,103 +137,16 @@ class Configuration {
-          permissions: {AdminPermission.listUsers, AdminPermission.removeUsers},
-        ),
-        AdminId(
-          email: 'pub-moderation-admin@dartlang-pub.iam.gserviceaccount.com',
-          oauthUserId: '108693445730271975989',
-          permissions: {AdminPermission.removePackage},
-        )
-      ],
-    );
-  }
+  factory Configuration.fromJson(Map<String, dynamic> json) =>
+      _$ConfigurationFromJson(json);
+  Map<String, dynamic> toJson() => _$ConfigurationToJson(this);
 
-  /// Create a configuration for development/staging deployment.
-  factory Configuration._dev() {
-    final projectId = 'dartlang-pub-dev';
-    return Configuration(
-      projectId: projectId,
-      packageBucketName: '$projectId--pub-packages',
-      dartdocStorageBucketName: '$projectId--dartdoc-storage',
-      popularityDumpBucketName: '$projectId--popularity',
-      searchSnapshotBucketName: '$projectId--search-snapshot',
-      backupSnapshotBucketName: '$projectId--backup-snapshots',
-      // TODO: Support finding search on localhost when envConfig.isRunningLocally
-      //       is true, this also requires running search on localhost.
-      searchServicePrefix: 'https://search-dot-$projectId.appspot.com',
-      storageBaseUrl: 'https://storage.googleapis.com/',
-      pubClientAudience: _pubClientAudience,
-      pubSiteAudience:
-          '621485135717-idb8t8nnguphtu2drfn2u4ig7r56rm6n.apps.googleusercontent.com',
-      adminAudience: 'https://pub.dev',
-      gmailRelayServiceAccount: null, // disable email sending
-      gmailRelayImpersonatedGSuiteUser: null, // disable email sending
-      uploadSignerServiceAccount:
-          'package-uploader-signer@dartlang-pub-dev.iam.gserviceaccount.com',
-      blockRobots: true,
-      productionHosts: envConfig.isRunningLocally
-          ? ['localhost']
-          : [
-              'dartlang-pub-dev.appspot.com',
-              '${envConfig.gaeService}-dot-dartlang-pub-dev.appspot.com',
-            ],
-      primaryApiUri: Uri.parse('https://dartlang-pub-dev.appspot.com'),
-      primarySiteUri: envConfig.isRunningLocally
-          ? Uri.parse('http://localhost:8080')
-          : Uri.parse(
-              'https://${envConfig.gaeVersion}-dot-dartlang-pub-dev.appspot.com',
-            ),
-      admins: [
-        AdminId(
-          oauthUserId: '111042304059633250784',
-          email: 'istvan.soos@gmail.com',
-          permissions: AdminPermission.values,
-        ),
-        AdminId(
-          oauthUserId: '117672289743137340098',
-          email: 'assigned-tags-admin@dartlang-pub-dev.iam.gserviceaccount.com',
-          permissions: {AdminPermission.manageAssignedTags},
-        )
-      ],
-    );
+  factory Configuration.fromYaml(final String path) {
+    final file = File(path);
+    final content = file.readAsStringSync();
+    final map =
+        json.decode(json.encode(loadYaml(content))) as Map<String, dynamic>;
+    return _$ConfigurationFromJson(map);
```

Comment:
Call `Configuration.fromJson` instead, we are using that pattern in other places.